  <launch>
    <arg name="map_size_x" value="40.0"/>
    <arg name="map_size_y" value="20.0"/>
    <arg name="map_size_z" value=" 3.0"/>

    <arg name="odom_topic" value="visual_slam/odom" />

<!--     <node pkg ="map_generator" name ="random_forest" type ="random_forest" output = "screen">    
        <param name="map/x_size"     value="$(arg map_size_x)" />
        <param name="map/y_size"     value="$(arg map_size_y)" />
        <param name="map/z_size"     value="$(arg map_size_z)" />
        <param name="map/resolution" value="0.1"/>        

        <param name="ObstacleShape/seed" value="1"/>
        
        <param name="map/obs_num"    value="70"/>
        <param name="ObstacleShape/lower_rad" value="0.5"/>
        <param name="ObstacleShape/upper_rad" value="0.7"/>
        <param name="ObstacleShape/lower_hei" value="0.0"/>
        <param name="ObstacleShape/upper_hei" value="3.0"/>        

        <param name="map/circle_num" value="70"/>        
        <param name="ObstacleShape/radius_l" value="0.7"/>        
        <param name="ObstacleShape/radius_h" value="0.5"/>        
        <param name="ObstacleShape/z_l" value="0.7"/>        
        <param name="ObstacleShape/z_h" value="0.8"/>        
        <param name="ObstacleShape/theta" value="0.5"/>        

        <param name="sensing/radius" value="5.0"/>        
        <param name="sensing/rate"   value="10.0"/>  
        
        <param name="min_distance" value="1.2"/>         
    </node> -->

    <node pkg ="map_generator" name ="dynamic_obs" type ="dynamic_obs" output = "screen">    
        <param name="map/x_size"     value="$(arg map_size_x)" />
        <param name="map/y_size"     value="$(arg map_size_y)" />
        <param name="map/z_size"     value="$(arg map_size_z)" />
        <param name="map/resolution" value="0.1"/>         

        <param name="sensing/radius" value="5.0"/>        
        <param name="sensing/rate"   value="10.0"/>  

        <param name="dynamic/x" value="-7.0"/>
        <param name="dynamic/z" value="2"/>
        <param name="dynamic/width" value="1.0"/>
        <param name="dynamic/height" value="0.2"/>
        <param name="dynamic/vel" value="2" />
        <param name="dynamic/num" value="2" />
    </node>    
  </launch>
